src/NosCore.Core/Serializing/PacketFactory.cs

```diff
@@ -302,8 +302,8 @@ private static IList DeserializeSimpleList(string currentValues, Type genericLis
                 var subPacketTypePropertiesCount = subpacketSerializationInfo.Value.Count;
 
                 // check if the amount of properties can be serialized properly
-                if ((splittedSubpacketParts.Count + (includesKeepAliveIdentity ? 1 : 0))
-                    % subPacketTypePropertiesCount
+                if (((splittedSubpacketParts.Count - (currentIndex.Value + 1 + (includesKeepAliveIdentity ? 1 : 0)))
```

Comment:
why ?

Reply:
Must add tests on another pull request, but this is because subpacketparts were not counted correctly